earn/src/components/borrow/modal/BorrowModal.tsx

```diff
@@ -1,19 +1,183 @@
+  UNABLE_TO_BORROW,
+  PENDING,
+  READY,
+}
+
+function getConfirmButton(state: ConfirmButtonState, token: Token): { text: string; enabled: boolean } {
+  switch (state) {
+    case ConfirmButtonState.UNABLE_TO_BORROW:
+      return {
+        text: `Unable to borrow ${token.ticker}`,
+        enabled: false,
+      };
+    case ConfirmButtonState.PENDING:
+      return { text: 'Pending', enabled: false };
+    case ConfirmButtonState.READY:
+      return { text: 'Confirm', enabled: true };
+    default:
+      return { text: 'Confirm', enabled: false };
+  }
+}
+
+type BorrowButtonProps = {
+  marginAccount: MarginAccount;
+  userAddress: string;
+  borrowToken: Token;
+  borrowAmount: Big;
+  shouldProvideAnte: boolean;
+  setIsOpen: (open: boolean) => void;
+  setPendingTxn: (result: SendTransactionResult | null) => void;
+};
+
+function BorrowButton(props: BorrowButtonProps) {
+  const { marginAccount, userAddress, borrowToken, borrowAmount, shouldProvideAnte, setIsOpen, setPendingTxn } = props;
+  const { activeChain } = useContext(ChainContext);
+
+  const [isPending, setIsPending] = useState(false);
+
+  const isBorrowingToken0 = borrowToken.address === marginAccount.token0.address;
+
+  const amount0Big = isBorrowingToken0 ? borrowAmount : new Big(0);
+  const amount1Big = isBorrowingToken0 ? new Big(0) : borrowAmount;
+
+  const marginAccountInterface = new ethers.utils.Interface(MarginAccountABI);
+  const encodedData = marginAccountInterface.encodeFunctionData('borrow', [
+    amount0Big.toFixed(),
+    amount1Big.toFixed(),
+    userAddress,
+  ]);
+
+  const { config: removeCollateralConfig } = usePrepareContractWrite({
+    address: marginAccount.address,
+    abi: MarginAccountABI,
+    functionName: 'modify',
+    args: [ALOE_II_SIMPLE_MANAGER, encodedData, [false, false]],
+    overrides: { value: shouldProvideAnte ? ANTE + 1 : undefined },
+    enabled: !!userAddress && borrowAmount.gt(0),
+    chainId: activeChain.id,
+  });
+  const removeCollateralUpdatedRequest = useMemo(() => {
+    if (removeCollateralConfig.request) {
+      return {
+        ...removeCollateralConfig.request,
+        gasLimit: removeCollateralConfig.request.gasLimit.mul(GAS_ESTIMATE_WIGGLE_ROOM).div(100),
+      };
+    }
+    return undefined;
+  }, [removeCollateralConfig.request]);
+  const {
+    write: contractWrite,
+    isSuccess: contractDidSucceed,
+    isLoading: contractIsLoading,
+    data: contractData,
+  } = useContractWrite({
+    ...removeCollateralConfig,
+    request: removeCollateralUpdatedRequest,
+  });
+
+  useEffect(() => {
+    if (contractDidSucceed && contractData) {
+      setPendingTxn(contractData);
+      setIsPending(false);
+      setIsOpen(false);
+    } else if (!contractIsLoading && !contractDidSucceed) {
+      setIsPending(false);
+    }
+  }, [contractDidSucceed, contractData, contractIsLoading, setPendingTxn, setIsOpen]);
+
+  let confirmButtonState = ConfirmButtonState.READY;
+
+  if (false) {
```

Comment:
???

Reply:
I mentioned this in the PR description; I didn't add logic for computing the maximum amount a user can borrow.

Reply:
Ok, if that's what this is intended to catch, I don't think "Unable to borrow XYZ" is the best button text. Maybe something like "Borrow amount exceeds limit" or "Account unhealthy"

Reply:
I wanted to leave this in as a reminder that the logic is not implemented. Also, because I didn't add the logic I didn't make a very specific title for the button. One of the reasons I didn't make it "account unhealthy" is because that doesn't factor in the case where there isn't enough supply.

Reply:
I'm not sure I follow

Reply:
It doesn't matter what we set it to since that button state isn't being used in this PR. I feel that once we add in the logic to handle these states, we can add buttons for the corresponding states.

Reply:
Then why have that enum value at all? If you're going to have it, the corresponding button text should be as correct as possible. If you _really_ don't want to do that here, then please make an issue about it so we don't forget.

Reply:
I can remove it